app/src/main/res/layout/activity_chat.xml

```diff
@@ -20,14 +20,41 @@
             android:layout_alignParentTop="true"
             android:scrollbars="vertical" />
 
+        <EditText
+            android:layout_width="match_parent"
+            android:id="@+id/chatSearchInput"
+            android:textColor="@color/md_white_1000"
+            android:layout_height="40dp"
+            android:layout_marginRight="@dimen/margin_extremely_large"
+            android:hint="Search Chat Messages"
+            android:layout_marginLeft="@dimen/margin_medium"
+            android:layout_marginTop="@dimen/margin_medium"
+            android:visibility="invisible"
+            android:paddingLeft="@dimen/padding_moderate"
+            android:background="@color/colorPrimaryDark"
+            android:textSize="@dimen/text_size_normal"/>
+
         <android.support.design.widget.FloatingActionButton
-            android:id="@+id/fabsetting"
+            android:id="@+id/searchChat"
             android:layout_width="wrap_content"
             android:layout_height="wrap_content"
             android:layout_alignParentEnd="true"
+            android:onClick="openChatSearch"
             android:layout_alignParentRight="true"
             android:layout_alignParentTop="true"
             android:layout_margin="@dimen/margin_moderate"
+            app:backgroundTint="@color/colorPrimary"
+            app:fabSize="mini"
+            app:srcCompat="@drawable/ic_open_search" />
+
+        <android.support.design.widget.FloatingActionButton
```

Comment:
Let the susi symbol be on top and the chat setting on the bottom

Reply:
It wont look nice. Usually the search bar appears at the top. So I preferred this positioning concept. If it is worthy, it can be changed later.